spark-dependencies/pom.xml

```diff
@@ -776,6 +818,14 @@
     </profile>
 
     <profile>
+      <!--only used for pyspark/sparkR needed spark dist is downloaded, no need to download auto-->
+      <id>spark-dist-cached</id>
```

Comment:
If we are changing the default behaviour (download&cache spark) we need to: 
 - update CI scripts 
 - update release docs 
as they rely on current behavior of spark been downloaded.

I would suggest we keep the default behavior: it downloads and caches spark in local FS and it's slow only once (download plugin have [ovewrite flag](https://github.com/maven-download-plugin/maven-download-plugin/blob/master/download-maven-plugin/src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java#L59) `false` by default) but allow developers, who do not need it at all, to explicitly opt-out this though the profie.